packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -145,16 +140,58 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
         }
     }
 
-    public componentDidMount() {
-        this.setTruncationState();
-    }
+    private renderPopover() {
+        const { children, preformatted } = this.props;
 
-    public componentDidUpdate() {
-        this.setTruncationState();
+        // <Popover> will always check the content's position on update
+        // regardless if it is open or not. This negatively affects perf due to
+        // layout thrashing. So instead we manage the popover state ourselves
+        // and mimic its popover target
+        if (this.state.isPopoverOpen) {
+            const popoverClasses = classNames(
+                Classes.TABLE_TRUNCATED_POPOVER,
+                preformatted ? Classes.TABLE_POPOVER_WHITESPACE_PRE : Classes.TABLE_POPOVER_WHITESPACE_NORMAL,
+            );
+            const popoverContent = <div className={popoverClasses}>{children}</div>;
+            const constraints = [
+                {
+                    attachment: "together",
+                    to: "window",
+                },
+            ];
+
+            return (
+                <Popover
+                    className={Classes.TABLE_TRUNCATED_POPOVER_TARGET}
+                    tetherOptions={{ constraints }}
+                    content={popoverContent}
+                    position={Position.BOTTOM}
+                    isOpen={true}
+                    onClose={this.handlePopoverClose}
+                    useSmartArrowPositioning={true}
+                >
+                    <Icon iconName="more" />
+                </Popover>
+            );
+        } else {
+            return (
+                <span className={Classes.TABLE_TRUNCATED_POPOVER_TARGET} onClick={this.handlePopoverOpen}>
```

Comment:
this reproduces `<Popover>`'s internal structure without calling any layout-triggering listeners